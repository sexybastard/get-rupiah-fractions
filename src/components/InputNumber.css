input.input-amount:focus {
  outline-width: 0;
}

::placeholder { /* Chrome, Firefox, Opera, Safari 10.1+ */
  opacity: 0.7;
}

.input-amount {
  flex: 1;
  padding-left: 32px;
  padding-right: 32px;
  border-radius: 4px;
  font-size: 20px;
  border-style: solid;
  border-width: 1px;
  border-color: #ccc;
  height: 54px;
}

.input-container {
  display: flex;
  flex-direction: row;
  margin-bottom: 12px;
  padding-left: 20px;
  padding-right: 20px;
}