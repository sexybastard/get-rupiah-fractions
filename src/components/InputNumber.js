import React, { Component } from 'react';
import { validate, getAmount, calculate } from '../utils';
import './InputNumber.css'

class InputNumber extends Component {
  constructor(props) {
    super(props);
  
    this.state = {
      amount: ''
    };
  }

  handleKeyPress = (e) => {
    if (e.key === 'Enter') {
      let validateResponse = validate(this.state.amount)
      if(validateResponse.isValid){
        this.props.setResult(calculate(getAmount(this.state.amount)), validateResponse)
      }
      else{
        this.props.setResult('', validateResponse)
      }
    }
  };

  handleChange = (e) => {
    this.setState({ amount: e.target.value })
  };

  render() {
    return (
      <div id="input-container" className="input-container">
        <input className="input-amount" placeholder="Input amount..." type="text" value={this.state.amount} onChange={this.handleChange} onKeyPress={this.handleKeyPress} />
      </div>
    );
  }
}

export default InputNumber;
