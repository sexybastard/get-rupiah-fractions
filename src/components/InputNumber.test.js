import React from 'react';
import { shallow } from 'enzyme';
import InputNumber from './InputNumber';

describe("Component: InputNumber should match snapshoot", () => {
  it("Initial case", () => {
    const wrapper = shallow(<InputNumber />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it("onChange event", () => {
    const wrapper = shallow(<InputNumber />);
    const event = {target: {value: "1"}};
    wrapper.find('#input-container input').first().simulate('change', event);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
})

describe("Component: InputNumber should have the correct className and content", () => {
  it("Ensure #input-container to have a correct className", () => {
    const wrapper = shallow(<InputNumber />);
    expect(wrapper.find('#input-container').prop('className')).toEqual('input-container')
  });

  it("Ensure #input-container input to have a correct className", () => {
    const wrapper = shallow(<InputNumber />);
    expect(wrapper.find('#input-container input').prop('className')).toEqual('input-amount')
  });

  it("Ensure #input-container input to have a correct initial value", () => {
    const wrapper = shallow(<InputNumber />);
    expect(wrapper.find('#input-container input').prop('value')).toEqual('')
  });
});

describe("Component: InputNumber should have the correct state evolution", () => {
  it("Ensure component to have a correct initial state", () => {
    const wrapper = shallow(<InputNumber />);
    expect(wrapper.state('amount')).toBe('')
    expect(wrapper.find('#input-container input').prop('value')).toBe('')
  });

  it("Ensure component to have a correct state after simulate event change", () => {
    const wrapper = shallow(<InputNumber />);
    const event = {target: {value: "1"}};
    wrapper.find('#input-container input').first().simulate('change', event);
    expect(wrapper.state('amount')).toBe(event.target.value)
    expect(wrapper.find('#input-container input').prop('value')).toBe(event.target.value)
  });
});