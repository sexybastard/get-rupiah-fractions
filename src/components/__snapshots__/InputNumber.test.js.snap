// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: InputNumber should match snapshoot Initial case 1`] = `
Array [
  <div
    className="input-container"
    id="input-container"
  >
    <input
      className="input-amount"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Input amount..."
      type="text"
      value=""
    />
  </div>,
]
`;

exports[`Component: InputNumber should match snapshoot onChange event 1`] = `
Array [
  <div
    className="input-container"
    id="input-container"
  >
    <input
      className="input-amount"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Input amount..."
      type="text"
      value="1"
    />
  </div>,
]
`;
